pkg/splunk/enterprise/util_test.go

```diff
@@ -1444,3 +1444,107 @@ func TestCopyFileToPod(t *testing.T) {
 		t.Errorf("Valid source and destination paths should not cause an error. Error: %s", err)
 	}
 }
+
+func TestCheckIfFileExistsOnPod(t *testing.T) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "splunk-stack1-0",
+			Namespace: "test",
+			Labels: map[string]string{
+				"controller-revision-hash": "v0",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/mnt/splunk-secrets",
+							Name:      "mnt-splunk-secrets",
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "mnt-splunk-secrets",
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: "test-secret",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create client and add object
+	c := spltest.NewMockClient()
+	// Add object
+	c.AddObject(pod)
+
+	filePathOnPod := "/init-apps/splunkFwdApps/testApp.tgz"
+
+	fileExists := checkIfFileExistsOnPod(c, "testNameSpace", pod.GetName(), filePathOnPod)
```

Comment:
is there a way to test the valid scenario too, where the file is present on pod?

Reply:
I don't think we can do that, ideally, this falls on the functional testing area. PoxExec command needs to be improved to cover this kind of testing, which is a backlog item(CSPL-582) for some time